src/main/kotlin/com/amazon/opendistroforelasticsearch/indexmanagement/indexstatemanagement/ManagedIndexRunner.kt

```diff
@@ -283,7 +308,9 @@ object ManagedIndexRunner : ScheduledJobRunner,
 
         if (updateResult && state != null && action != null && step != null && currentActionMetaData != null) {
             // Step null check is done in getStartingManagedIndexMetaData
-            step.preExecute(logger).execute().postExecute(logger)
+            runBlocking(InjectorContextElement(managedIndexConfig.id, settings, threadPool.threadContext, roles)) {
```

Comment:
Don't use runBlocking

Reply:
use withContext and do roleInject.close at the end of block.
@skkosuri-amzn alerting can change to this also